pinot-plugins/pinot-batch-ingestion/v0_deprecated/pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/mappers/SegmentCreationMapper.java

```diff
@@ -243,14 +257,15 @@ protected void map(LongWritable key, Text value, Context context)
     addAdditionalSegmentGeneratorConfigs(segmentGeneratorConfig, hdfsInputFile, sequenceId);
 
     _logger.info("Start creating segment with sequence id: {}", sequenceId);
-    SegmentIndexCreationDriver driver = new SegmentIndexCreationDriverImpl();
+    SegmentIndexCreationDriverImpl driver = new SegmentIndexCreationDriverImpl();
```

Comment:
Seems like we are breaking interface here, what' the reasoning for that? Either the api should be justified to be part of the interface, or the design is broken somehow.

Reply:
+1
It has been on my radar to add a columnar segment creation driver (for realtime), and this will break completely

Reply:
Added a class called `SchemaValidator` to validate Pinot schema and the input data schema